File: test/fixtures/ast-grep/no-empty-catch-java/ok.java
package fixtures;

import java.io.IOException;
import java.io.UncheckedIOException;

public class Ok {

    // OK: catch with real handling (wrap and rethrow)
    public void rethrows() {
        try {
            risky();
        } catch (IOException e) {
            throw new UncheckedIOException("risky failed", e);
        }
    }

    // OK: near-miss - comment plus a real statement must NOT be flagged
    public void handlesWithComment() {
        try {
            risky();
        } catch (IOException e) {
            // fall back to the cached value
            report(e);
        }
    }

    // OK: try/finally without any catch clause
    public void finallyOnly() {
        try {
            System.out.println("work");
        } finally {
            cleanup();
        }
    }

    private void risky() throws IOException {
        throw new IOException("boom");
    }

    private void report(Exception e) {
        System.out.println(e.getMessage());
    }

    private void cleanup() {
        System.out.println("cleanup");
    }
}

File: test/fixtures/ast-grep/no-empty-catch-js/ok.js
function risky() {}
function report(err) {}
const logger = { error(msg, err) {} };

// OK: catch with real handling (rethrow)
export function rethrows() {
  try {
    risky();
  } catch (e) {
    throw new Error(`risky failed: ${String(e)}`);
  }
}

// OK: catch that handles the error, with a comment alongside the statement
export function handlesWithComment() {
  try {
    risky();
  } catch (e) {
    // near-miss: comment plus a real statement must NOT be flagged
    report(e);
  }
}

// OK: try/finally without catch at all
export function finallyOnly() {
  try {
    risky();
  } finally {
    logger.error('done', null);
  }
}
